vispy/scene/widgets/axis.py

```diff
@@ -76,13 +78,20 @@ def link_view(self, view):
                 self._view_changed)
         self._linked_view = view
         view.scene.transform.changed.connect(self._view_changed)
+        self._tr = self.node_transform(self._linked_view.scene)
```

Comment:
Is `_view_changed` triggered/called when the transforms changed? Does this `self._tr` need to get updated because of that?

I'm forgetting everything associated with this, but iirc the performance issue is that `self._tr` gets rebuilt every time the view is changed. This includes transform attributes being updated where `self._tr` wouldn't change (the overall shader code is the same, right?) but `_view_changed` would still be called because it is a change to the transform that requires a redraw. However, if the actual transform object of something in the visual node graph changed (requiring shader recompilation/building) then `_view_changed` is called and `self._tr` should be updated. Does any of this sound reasonable?